### PR TITLE
add shared path to SSM parameters

### PR DESCRIPTION
**Pull request by konstl000**
[add yourself]: https://github.com/concourse/governance#individual-contributors
[PR requirements]: https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md#pull-request-requirements

## Changes proposed by this PR

As of now, it is not possible to configure shared paths for AWS SSM cred manager as opposed to other cred managers such as Vault. Since it is easy to implement and useful, I propose to add a new parameter `--aws-ssm-shared-path` similar to the Vault cred manager. A path prefix rather than template (as for AWS SecretsManager) makes sense since in this case no default is needed and this parameter can be optional (as with Vault). This PR provides all the changes needed for this functionality.

## Release Note

*  Added `--aws-ssm-shared-path` to configure shared secret paths for AWS SSM cred manager similarly to the one for Vault.

[imperative mood]: https://chris.beams.io/posts/git-commit/#imperative

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -226,6 +226,8 @@ require (
 	github.com/go-openapi/jsonreference v0.20.1 // indirect
 	github.com/go-openapi/swag v0.22.3 // indirect
 	github.com/go-task/slim-sprig v0.0.0-20210107165309-348f09dbbbc0 // indirect
+	github.com/gobuffalo/envy v1.7.0 // indirect
+	github.com/gobuffalo/packd v0.3.0 // indirect
```
Comment (xtremerui) [inline go.mod:230]: These don't look like needed for the PR. Could you remove them? Thank you!

Comment (konstl000) [reply go.mod]: Done. Now there are somehow complaints about signoff and commit history is somehow botched. Should I maybe just re-introduce my changes to the current master and re-open the PR?

Comment (konstl000) [reply go.mod]: Should be fine now

**Review by taylorsilva — COMMENTED** [review]
Gave this a quick look and looks good! Gonna dig into this a bit more when I have more time and then approve if I have no feedback. Thanks for the PR!

**Comment by konstl000** [comment]
> Gave this a quick look and looks good! Gonna dig into this a bit more when I have more time and then approve if I have no feedback. Thanks for the PR!

It's been in total already about 9 months for many of the PRs hanging around. There also have been multiple releases during this time. Are there some issues that prevent merging or do you consider if you should merge at all?

**Comment by xtremerui** [comment]
@taylorsilva ^^

**Comment by taylorsilva** [comment]
sorry! Got busy at work ☹️

**Review by taylorsilva — APPROVED** [review]
Thanks again for the PR! Sorry for the delay reviewing. Concourse ain't my full time job anymore and my actual job has kept me busy. Thanks for the ping @xtremerui

**Comment by konstl000** [comment]
No problem, just wanted to know where we are at :-)
Thanks for a quick response and merge!

**Pull request merged**